### PR TITLE
Add longer descriptions to impls in `flags`

### PR DESCRIPTION
**Pull request by keckler**
## What is the change?

Just adds descriptions to the implementation tags in the `flags` module.

## Why is the change being made?

Quality.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [ ] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/reactor/flags.py [hunk 1]
```diff
@@ -280,29 +228,78 @@ class Flags(Flag):
...
             :implements: R_ARMI_FLAG_TO_STR
+
+            This converts the representation of a bunch of flags from ``typeSpec``,
+            which might look like ``Flags.A|B``,
+            into a string with spaces in between the flag names, which would look
+            like  ``'A B'``. This is done via nesting string splitting and replacement
+            actions.
         """
-        return _toString(cls, typeSpec)
+        return str(typeSpec).split("{}.".format(cls.__name__))[1].replace("|", " ")
```
Comment (john-science) [inline armi/reactor/flags.py:302]: Is this where the error is happening?

Comment (john-science) [reply armi/reactor/flags.py:302]: Let's not make any code changes.  

In the last months ARMI has had (I think) zero code changes.  Let's stick with that for 2 more days.

Comment (linconnor) [reply armi/reactor/flags.py]: Although I didn't see anything wrong with Chris changes (line 120: def __fromStringGeneral(cls, typeSpec, updateMethod)), but as John suggested, keep the code as it is.

**Review by linconnor — APPROVED** [review]
The added impls info looks good.

File: armi/reactor/flags.py [hunk 2]
```diff
@@ -290,6 +290,17 @@ def fromString(cls, typeSpec):
         .. impl:: Retrieve flag from a string.
             :id: I_ARMI_FLAG_TO_STR0
             :implements: R_ARMI_FLAG_TO_STR
+
+            For a string passed as ``typeSpec``, first converts the whole string
+            to uppercase.
+            Then tries to parse the string for any special phrases, as defined
+            in the module dictionary ``_CONVERSIONS``, and converts those phrases
+            to flags directly.
```
Comment (john-science) [inline armi/reactor/flags.py:298]:
```suggestion
            For a string passed as ``typeSpec``, first converts the whole string
            to uppercase. Then tries to parse the string for any special phrases, as
            defined in the module dictionary ``_CONVERSIONS``, and converts those
            phrases to flags directly.
```

Comment (john-science) [reply armi/reactor/flags.py:298]: One line of text only had two words. It was just a bit awkward, so I... justified the text a bit.

File: armi/reactor/flags.py [hunk 3]
```diff
@@ -290,6 +290,17 @@ def fromString(cls, typeSpec):
...
+
+            Then splits the remaining string into separate words based on the presence
```
Comment (john-science) [inline armi/reactor/flags.py:300]:
```suggestion
            Then it splits the remaining string into separate words based on the presence
```

File: armi/reactor/flags.py (hunk 2 above)
Comment (linconnor) [reply armi/reactor/flags.py:298]: Right, I miss this part.

**Review by john-science — APPROVED** [review]

**Pull request merged**